src/libcore/mem.rs

```diff
@@ -611,3 +611,35 @@ pub const POST_DROP_USIZE: usize = POST_DROP_U64 as usize;
 pub unsafe fn transmute_copy<T, U>(src: &T) -> U {
     ptr::read(src as *const T as *const U)
 }
+
+/// Returns the value of the discriminant for the enum variant in `v`.
+///
+/// The returned discriminant is cast to `u64`, no matter what the actual
+/// representation is. Note that if the enum's representation is signed, then
+/// the discriminant values returned by this function will not compare in the
+/// same order as actual enum variants would under `#[derive(PartialOrd)]`.
+/// 
+/// If `T` is not an enum, the return value is unspecified.
+///
+/// # Example
+///
+/// This can be used to compare enums that carry data, while disregarding
+/// the actual data:
+///
+/// ```
+/// use std::mem;
+///
+/// enum Foo { A(String), B(i32) }
+///
+/// # fn main() {
+/// assert!(mem::discriminant(Foo::A("bar")) != mem::discriminant(Foo::A("baz")));
```

Comment:
Yeah, I just noticed that and fixed it.

On Tue, Jul 12, 2016 at 2:19 PM, Alex Crichton <notifications@github.com>
wrote:

> In src/libcore/mem.rs
> <https://github.com/rust-lang/rust/pull/34785#discussion_r70492383>:
>
> > +/// same order as actual enum variants would under `#[derive(PartialOrd)]`.
> > +///
> > +/// If `T` is not an enum, the return value is unspecified.
> > +///
> > +/// # Example
> > +///
> > +/// This can be used to compare enums that carry data, while disregarding
> > +/// the actual data:
> > +///
> > +/// ```
> > +/// use std::mem;
> > +///
> > +/// enum Foo { A(String), B(i32) }
> > +///
> > +/// # fn main() {
> > +/// assert!(mem::discriminant(Foo::A("bar")) != mem::discriminant(Foo::A("baz")));
>
> I think these examples may not compile, the discriminants should be the
> same here, right? Additionally, shouldn't a shared reference be passed in?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rust-lang/rust/pull/34785/files/3a88d8811ae23af836e8ed4349e4a96656303878#r70492383>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAC3n1nIqX5ydVTZ2ue3rNp94IPEA-khks5qU9qggaJpZM4JKnnt>
> .
>